src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. This can be seen as a code quality issue since it encapsulates the creation logic within the constructor.